src/cosmos/CosmosEngine.ts

```diff
@@ -303,44 +300,49 @@ export class CosmosEngine extends CurrencyEngine<
   processCosmosTransaction(
     txidHex: string,
     date: number,
-    event: TransferEvent,
-    tx: TxResponse
+    signedTx: string,
+    coin: CosmosCoin,
+    memo: string,
+    height: number,
+    fee?: Fee
   ): void {
-    const { height, tx: txRaw } = tx
-    const signedTx = base16.stringify(txRaw)
-    const {
-      authInfo: { fee },
-      body: { memo }
-    } = decodeTxRaw(txRaw)
+    const { amount, denom } = coin
+
+    const isMainnet = this.networkInfo.nativeDenom === denom
+
+    const currencyCode =
+      this.networkInfo.nativeDenom === denom
+        ? this.currencyInfo.currencyCode
+        : this.allTokensMap[denom] != null
+        ? this.allTokensMap[denom].currencyCode
+        : undefined
+    if (currencyCode == null) return
 
     let networkFee = this.networkInfo.defaultTransactionFee?.amount ?? '0'
     if (fee != null) {
       const { amount } = fee
       const networkFeeCoin = safeAddCoins([
-        coin('0', this.networkInfo.nativeDenom),
+        { amount: '0', denom: this.networkInfo.nativeDenom },
         ...amount
       ])
       networkFee = add(networkFee, networkFeeCoin.amount)
     }
 
-    const { coin: eventCoin, recipient, sender } = event
-
-    if (
-      (this.walletInfo.keys.bech32Address !== recipient &&
-        this.walletInfo.keys.bech32Address !== sender) ||
-      this.networkInfo.nativeDenom !== eventCoin.denom
-    ) {
-      return
-    }
-    const isSend = this.walletInfo.keys.bech32Address === sender
+    const isSend = lt(amount, '0')
```

Comment:
technically `isSend` is whether the users keys signed this tx. Can you tell based on the tx if it was signed by the user?